tests/util/pilot_server.go

```diff
@@ -196,16 +189,19 @@ func setup() error {
 	MockPilotSecurePort, _ = strconv.Atoi(port)
 
 	// Wait a bit for the server to come up.
-	// TODO(nmittler): Change to polling health endpoint once https://github.com/istio/istio/pull/2002 lands.
-	time.Sleep(time.Second)
-
-	return nil
-}
-
-// Teardown will cleanup the temp dir and remove the test data.
-func Teardown() {
-	close(stop)
+	err = wait.Poll(500*time.Millisecond, 5*time.Second, func() (bool, error) {
+		client := &http.Client{Timeout: 1 * time.Second}
+		resp, err := client.Get(MockPilotURL + "/ready")
+		if err != nil {
```

Comment:
If the port hasn't been opened yet (suppose the server at line 164 is still starting), then the 'Get' call will return error code, right?

Reply:
If the port is not opened yet, we will get an error like connection refused.

Reply:
In that case, it will return false. Thought that it is to wait until the server is up. I see that poll wait 500 ms before the first client.Get() call. Maybe that's enough for the server to have the port opened. Otherwise, it'd return false prematurely.